extensions/guacamole-auth-ldap/src/main/java/org/apache/guacamole/auth/ldap/user/UserService.java

```diff
@@ -85,11 +85,20 @@ private void putAllUsers(Map<String, User> users, LDAPConnection ldapConnection,
 
         try {
 
+            // Build a filter using the configured or default user search filter
+            // to find all user objects in the LDAP tree
+            StringBuilder userSearchFilter = new StringBuilder();
+            userSearchFilter.append("(&");
+            userSearchFilter.append(confService.getUserSearchFilter());
+            userSearchFilter.append("(" + escapingService.escapeLDAPSearchFilter(usernameAttribute) + "=*)");
```

Comment:
If using a `StringBuilder` (good), you shouldn't be doing string concatenation with `+`. The inline concatenation here will actually result in Java creating a temporary `StringBuilder` for the concatenation operation.

You should either do the whole thing with a single line of chained `+`:

    "(&" + confService.getUserSearchFilter() + "(" + escapingService.escapeLDAPSearchFilter(usernameAttribute) + "=*))"

or use only the `StringBuilder` you've created:

    StringBuilder userSearchFilter = new StringBuilder();
    userSearchFilter.append("(&");
    userSearchFilter.append(confService.getUserSearchFilter());
    userSearchFilter.append("(");
    userSearchFilter.append(escapingService.escapeLDAPSearchFilter(usernameAttribute));
    userSearchFilter.append("=*))");

FYI: In the case of the construction of the other query below (```StringBuilder ldapQuery = ...```), the use of a `StringBuilder` is absolutely necessary, since parts of the concatenation operation are conditional, and using Java's `+` operator would result in unnecessary creation of temporary `String` and `StringBuilder` objects for intermediate results.

In this case, you're safe either way, but using both is wasteful.